boot/freeldr/freeldr/ntldr/arch/i386/winldr.c

```diff
@@ -600,14 +600,23 @@ WinLdrSetProcessorContext(void)
      * Longhorn/Vista reports LimitLow == 0x0fff == MM_PAGE_SIZE - 1, whereas
      * Windows 7+ uses larger sizes there (not aligned on a page boundary).
      */
-#if 1
+#if 0
     /* Server 2003 way */
     KiSetGdtEntryEx(KiGetGdtEntry(pGdt, KGDT_R0_PCR), (ULONG32)Pcr, 0x1,
                     TYPE_DATA, DPL_SYSTEM, TRUE, 2);
 #else
-    /* Vista+ way */
-    KiSetGdtEntry(KiGetGdtEntry(pGdt, KGDT_R0_PCR), (ULONG32)Pcr, MM_PAGE_SIZE - 1,
-                  TYPE_DATA, DPL_SYSTEM, 2);
+    if (OperatingSystemVersion >= _WIN32_WINNT_VISTA)
```

Comment:
Regarding this GDT granularity bit "problem" on Vista @maharmstone , this is now taking this form. Could you please confirm to me whether the so-called "Server 2003 way" that is below could actually work in Vista, or do we really need to use the "Vista+ way" ?

Reply:
Sorry for the delay, I've been busy with other things. I've just tried your version, and no it doesn't work - it reboots without even a blue screen. Replacing your preprocessor stuff with an if makes it work, though.

I've just pushed a rebased version based on the current master.